src/poetry/utils/pip.py

```diff
@@ -43,6 +43,9 @@ def pip_install(
             )
         args.append("-e")
 
+    if subdirectory:
+        path = Path(f"file:{str(path)}#subdirectory={subdirectory}")
```

Comment:
A file: URL isn’t a `Path`, and on some systems the `Path` constructor modifies the string in system-specific ways. Since we’re about to convert this right back to `str`, we should rearrange this to leave the file: URL as `str`.